pom.xml

```diff
@@ -150,6 +150,10 @@
                     <java>
                         <googleJavaFormat>
                             <style>GOOGLE</style>
+                            <!-- Last version that supports Java 8, the plugin does handle this internally
+                                 but it will give different versions based off of the JVM version, not good for CI
+                             -->
+                            <version>1.7</version>
```

Comment:
I'm not sure I follow what is going on here.
Without this the spotless tool uses a java version dependant formatter, which you pointed out, but the formatter is compiled with java8 [regardless of the choice of version](https://github.com/google/google-java-format/blob/google-java-format-1.9/pom.xml#L97)  so why would we not want this to be the latest which at the time of writing is `1.9`? 
is this a case where the the upstream is using `--source` and `--target` but not `--release` and as such using not available at runtime method?

Reply:
> the reason is because the rules are different

so an upgrade to Java11 requires a complete reformat?
In which case I would suggest we just go with the latest rules becuase we are going to have to change at some point?

Reply:
it's very minor, I only had one class complaining, java 8 rules required me moving a chained method, something like:

```
blah
  .get()
  .doSomething()
```

up to this on java 8:
```
blah.get()
  .doSomething()
```

while it was fine on java 11.

Reply:
I think I would be tempted to say use `1.9` and apply the required code patch.  @bitwiseman thoughts?

Reply:
won't that mean that locally you will be required to run java 11? (no issue with that but it should probably be enforced?)

Reply:
https://github.com/google/google-java-format/releases/tag/google-java-format-1.8

> The minimum supported runtime version is now JDK 11.

uhuhuuu... 

https://github.com/google/google-java-format/blob/google-java-format-1.9/core/pom.xml#L250-L251

okey dokey.....

Reply:
@timja 

I know it is arbitrary, but I'd like to avoid locking the formatter version.  I'd rather put a note in the CONTRIBUTING.md and the PR template saying:

> We recommend you run Java 11 in while developing this plugin. We use Google Formatter to automate and enforce formatting.  It requires Java 11+.  You can still build and test using Java 8, but you will not get formatting checks. CI will inform you of formatting errors. You can fix them manually or switch to Java 11 and run `spotless:apply`. 

And then add a profile to the pom.xml that skips checking on Java 8. 

@jtnord 
> Apply the required code patch

What do you mean here?

Reply:
> @timja 
> 
> I know it is arbitrary, but I'd like to avoid locking the formatter version.  I'd rather put a note in the CONTRIBUTING.md and the PR template saying:
> 
> > We recommend you use Java 11 in while developing this plugin. The project uses Google Formatter to automate and enforce formatting.  It requires Java 11+.  You can still build and test using Java 8, but you will not get formatting checks. The CI build on Java 11 will inform you of formatting errors. You can fix them manually or switch to Java 11 and run `spotless:apply`.
> 
> And then add a profile to the pom.xml that skips checking on Java 8. 


I don't think the plugin pom is correctly so for that, I wouldn't recommend using java 11 until we release java 11 bytecode

> @jtnord 
> > Apply the required code patch
> 
> What do you mean here? 

Change everything that was needed in the formatter for the newer format...

(Which is one line of code iiuc)